src/core/src/lib/components/formly.field.ts

```diff
@@ -283,12 +283,9 @@ export class FormlyField implements DoCheck, OnInit, OnChanges, AfterContentInit
     if (!field) {
       return () => {};
     }
-
-    const subscribes = [observeDeep(field, ['props'], () => field.options.detectChanges(field))];
-
-    if (field.options) {
-      subscribes.push(observeDeep(field.options, ['formState'], () => field.options.detectChanges(field)));
-    }
+    
+    const propsObserver = observe(field, ['props'], () => field.options.detectChanges(field));
```

Comment:
I think we should keep `observeDeep`. if there is a specific case where the perf drop we need to provide a fix for. `observeDeep` instead.